spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java

```diff
@@ -426,8 +427,10 @@ else if (this.autoStartup != null) {
 				.acceptIfNotNull(this.phase, instance::setPhase)
 				.acceptIfNotNull(this.applicationContext, instance::setApplicationContext)
 				.acceptIfNotNull(this.applicationEventPublisher, instance::setApplicationEventPublisher)
-				.acceptIfNotNull(endpoint.getGroupId(), instance.getContainerProperties()::setGroupId)
-				.acceptIfNotNull(endpoint.getClientIdPrefix(), instance.getContainerProperties()::setClientId)
+				.acceptIfCondition(StringUtils.hasText(endpoint.getGroupId()), endpoint.getGroupId(),
```

Comment:
I see there is a convenient `acceptIfHasText()` instead of this combination.
What am I missing?

Reply:
Oops... forgot about that.